bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/WeatherApi.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.netatmo.internal.api;
+
+import static org.openhab.binding.netatmo.internal.api.data.NetatmoConstants.*;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+
+import javax.ws.rs.core.UriBuilder;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.netatmo.internal.api.data.NetatmoConstants.MeasureClass;
+import org.openhab.binding.netatmo.internal.api.dto.NAMain;
+import org.openhab.binding.netatmo.internal.api.dto.NAMain.NAStationDataResponse;
+import org.openhab.binding.netatmo.internal.api.dto.NAMeasureBodyElem;
+
+/**
+ *
+ * @author Gaël L'hopital - Initial contribution
+ *
+ */
+
+@NonNullByDefault
+public class WeatherApi extends RestManager {
+    public WeatherApi(ApiBridge apiClient) {
+        super(apiClient, org.openhab.binding.netatmo.internal.api.data.NetatmoConstants.FeatureArea.WEATHER);
+    }
+
+    /**
+     *
+     * The method getStationsData Returns data from a user&#x27;s Weather Stations (measures and device specific
+     * data).
+     *
+     * @param deviceId Id of the device you want to retrieve information of (optional)
+     * @param getFavorites Whether to include the user&#x27;s favorite Weather Stations in addition to the user&#x27;s
+     *            own Weather Stations (optional, default to false)
+     * @return NAStationDataResponse
+     * @throws NetatmoException If fail to call the API, e.g. server error or cannot deserialize the
+     *             response body
+     */
+    public NAStationDataResponse getStationsData(@Nullable String deviceId, boolean getFavorites)
+            throws NetatmoException {
+        UriBuilder uriBuilder = getApiUriBuilder(SPATH_GETSTATION, PARM_DEVICEID, deviceId, PARM_FAVORITES,
+                getFavorites);
+        NAStationDataResponse response = get(uriBuilder, NAStationDataResponse.class);
+        return response;
+    }
+
+    public NAMain getStationData(String deviceId) throws NetatmoException {
+        ListBodyResponse<NAMain> answer = getStationsData(deviceId, true).getBody();
+        NAMain station = answer.getElement(deviceId);
+        if (station != null) {
+            return station;
+        }
+        throw new NetatmoException(String.format("Unexpected answer cherching device '%s' : not found.", deviceId));
```

Comment:
searching :)

Reply:
Loool :)

Reply:
Only French people can understand why it is funny lol